src/derivative_operators/derivative_operator_functions.jl

```diff
@@ -38,52 +38,49 @@ for MT in [2,3]
             ndimsM = ndims(M)
             @assert N <= ndimsM
 
-            # Respahe x_temp for NNlib.conv!
-            new_size = Any[size(x_temp)...]
+            # Determine padding for NNlib.conv!
             bpc = A.boundary_point_count
-            setindex!(new_size, new_size[N]- 2*bpc, N)
-            new_shape = []
-            for i in 1:ndimsM
-                if i != N
-                    push!(new_shape,:)
-                else
-                    push!(new_shape,bpc+1:new_size[N]+bpc)
-                end
-             end
-             _x_temp = reshape(view(x_temp, new_shape...), (new_size...,1,1))
+            pad = zeros(Int64,ndimsM)
+            pad[N] = bpc
+
+            # Reshape x_temp for NNlib.conv!
+            _x_temp = reshape(x_temp, (size(x_temp)...,1,1))
 
             # Reshape M for NNlib.conv!
             _M = reshape(M, (size(M)...,1,1))
-            s = A.stencil_coefs
-            sl = A.stencil_length
 
             # Setup W, the kernel for NNlib.conv!
+            s = A.stencil_coefs
+            sl = A.stencil_length
             Wdims = ones(Int64, ndims(_x_temp))
             Wdims[N] = sl
             W = zeros(Wdims...)
             Widx = Any[Wdims...]
             setindex!(Widx,:,N)
-            W[Widx...] = s ./ A.dx^A.derivative_order # this will change later 
-            cv = DenseConvDims(_M, W)
+            W[Widx...] = s
 
+            cv = DenseConvDims(_M, W, padding=pad)
             conv!(_x_temp, _M, W, cv)
 
             # Now deal with boundaries
-            dimsM = [axes(M)...]
-            alldims = [1:ndims(M);]
-            otherdims = setdiff(alldims, N)
-
-            idx = Any[first(ind) for ind in axes(M)]
-            itershape = tuple(dimsM[otherdims]...)
-            nidx = length(otherdims)
-            indices = Iterators.drop(CartesianIndices(itershape), 0)
-
-            setindex!(idx, :, N)
-            for I in indices
-                Base.replace_tuples!(nidx, idx, idx, otherdims, I)
-                convolve_BC_left!(view(x_temp, idx...), view(M, idx...), A)
-                convolve_BC_right!(view(x_temp, idx...), view(M, idx...), A)
+            if bpc > 0
+                dimsM = [axes(M)...]
+                alldims = [1:ndims(M);]
+                otherdims = setdiff(alldims, N)
+
+                idx = Any[first(ind) for ind in axes(M)]
```

Comment:
Why `Any`?

Reply:
@ChrisRackauckas at line 76 I am calling setindex!(idx, :, N), and Colon is not a subtype of Number

Reply:
oh yes.

Reply:
This might ruin inference though. Maybe `ntuple` is more appropriate? That can be optimized later